apps/docs/src/.vitepress/components/ComponentCard.vue

```diff
@@ -18,7 +14,21 @@ export type ComponentCardProps = {
    * @example "Q2 2024"
    */
   dueDate?: string;
-};
+} & (
+  | {
+      /** Component status. */
+      status: Extract<ComponentStatus, "in-progress" | "planned">;
+      /** Link to the components storybook. */
+      href?: string;
+    }
+  | {
+      status: Exclude<ComponentStatus, "in-progress" | "planned">;
```

Comment:
```suggestion
      status: Exclude<ComponentStatus, "implemented">;
      /** Link to the components storybook. */
      href?: string;
    }
  | {
      status: Extract<ComponentStatus, "implemented">;
```

Isn't it easier to just pick the implemented status instead of excluding all others?